### PR TITLE
Trigger to generate Windows boot files

### PR DESCRIPTION
**Pull request by tpw56j**

### CODE REVIEW

**Review by hbokh — APPROVED** [review]
Seems to be well written with good examples.

**Comment by dansou901** [comment]
I'll add my review as soon as I have time for it, please don't merge before that.

File: _posts/2020-12-04-windows.md [hunk 1]
```diff
@@ -65,21 +65,22 @@ and copy the Windows distributions there:
 Copy the following files to the distributions directories (for Windows 7 and newer):
```
Comment (dansou901) [inline _posts/2020-12-04-windows.md:65]: distributions directory (it's just one, isn't it?)

File: _posts/2020-12-04-windows.md [hunk 2]
```diff
@@ -0,0 +1,578 @@
...
+
+Despite the success of Microsoft in creating tools for integrating with Linux, in my opinion, Linux is more suitable for such a platform thanks to the wonderful project Cobbler. Cobbler is incredibly flexible and can be tailored to suit any need. Using templates and Cobbler profiles with the ability to inherit (parent profile), you can change a couple of lines in the template, you can very quickly create an installation option adapted to the needs of a particular user without having to depend on the capabilities of the closed proprietary code.
+
+Cobbler has an excellent architectural feature of expanding functionality by writing triggers that do not affect the main code. We used this opportunity to prepare the necessary files for a Windows installation.
+
+Linux currently has all the necessary tools to prepare these files. All that remains is to combine these tools in the trigger code to get an open source infrastructure for deploying Windows without using proprietary Microsoft tools.
+
+Trigger sync_post_wingen.py does:
```
Comment (dansou901) [inline _posts/2020-12-04-windows.md:18]: without "does"

File: _posts/2020-12-04-windows.md [hunk 3]
```diff
@@ -0,0 +1,578 @@
...
+- some of the files are created from standard ones (`pxeboot.n12`, `bootmgr.exe`) by directly replacing one sting with another directly in the binary
+- in the process of changing the bootmgr.exe file, the checksum of the PE file will change and it needs to be recalculated. The trigger does this with `python-pefile`
+- `python3-hivex` is used to modify Windows boot configuration data (BCD). For pxelibux distro boot_loader in BCD, paths to winpe.wim and boot.sdi are generated as `/winos/<distro_name>/boot`, and for iPXE - `\Boot`.
```
Comment (dansou901) [inline _posts/2020-12-04-windows.md]: pxelinux instead of pxelibux

**Review by dansou901 — CHANGES_REQUESTED** [review]
see inline comments

File: _posts/2020-12-04-windows.md (hunk 1 above)
Comment (tpw56j) [reply _posts/2020-12-04-windows.md]: I meant that you need to take the appropriate files for Win 10 and copy them to winos/Win10_EN-x64, and copy the files for Win 7 to winos/Win7_EN-x64. Files for Win 10 may not work for Win 7.

```
winos |
      + Win10_EN-x64
                     + pxeboot.n12
                     | bootmgr.exe
                       < .. >
                       
      + Win7__EN-x64
                     + pxeboot.n12
                     | bootmgr.exe
                       < .. >
```
Maybe this is a bad phrase or just need more clarification?

File: _posts/2020-12-04-windows.md (hunk 3 above)
Comment (tpw56j) [reply _posts/2020-12-04-windows.md]: Thanks for finding these errors! I have corrected them.

File: _posts/2020-12-04-windows.md [hunk 4]
```diff
@@ -0,0 +1,578 @@
...
+
+Despite the success of Microsoft in creating tools for integrating with Linux, in my opinion, Linux is more suitable for such a platform thanks to the wonderful project Cobbler. Cobbler is incredibly flexible and can be tailored to suit any need. Using templates and Cobbler profiles with the ability to inherit (parent profile), you can change a couple of lines in the template, you can very quickly create an installation option adapted to the needs of a particular user without having to depend on the capabilities of the closed proprietary code.
+
+Cobbler has an excellent architectural feature of expanding functionality by writing triggers that do not affect the main code. We used this opportunity to prepare the necessary files for a Windows installation.
+
+Linux currently has all the necessary tools to prepare these files. All that remains is to combine these tools in the trigger code to get an open source infrastructure for deploying Windows without using proprietary Microsoft tools.
+
+Trigger sync_post_wingen.py does:
+- some of the files are created from standard ones (`pxeboot.n12`, `bootmgr.exe`) by directly replacing one sting with another directly in the binary
```
Comment (dansou901) [inline _posts/2020-12-04-windows.md]: string instead of sting

**Review by dansou901 — CHANGES_REQUESTED** [review]
just one more which slipped through...

File: _posts/2020-12-04-windows.md (hunk 4 above)
Comment (tpw56j) [reply _posts/2020-12-04-windows.md]: @dansou901 I haven't studied git very well yet, and maybe my commit https://github.com/tpw56j/cobbler.github.io/commit/42552f625495ed06896763226e474d9e462c914f was not included in the PR?

**Review by SchoolGuy — APPROVED** [review]
Can't test but it definitely is more then I could do with my knowledge.

**Comment by SchoolGuy** [comment]
@dansou901 Your call now.

**Review by dansou901 — APPROVED** [review]
LGTM now, at least for grammar and spelling. For the rest, I can't really tell and I also can't try it out now.

**Review by SchoolGuy — CHANGES_REQUESTED** [review]
It would be absolute superb if you could include the windows changes in the `distro_signatures.json` file in here so folks don't lose that if they update the signatures from up here!

**Comment by tpw56j** [comment]
@SchoolGuy I have included the changes for windows in the distro_signatures.json file: https://github.com/cobbler/cobbler/pull/2466/commits/60c7a9e979019ff56f8c3874a6afc720774a15cc

**Comment by tpw56j** [comment]
@SchoolGuy Can then remove this item from here?

**Comment by SchoolGuy** [comment]
@tpw56j Cobbler has the possibility to update the signatures without a version update. Thus we need both. But I will do it myself then. Its not a problem.

**Pull request merged**